### PR TITLE
User Events support in OpenWhisk Standalone mode

### PR DESCRIPTION
**Pull request by chetanmeh**
Adds support for launching User Event service along with preconfigured Grafana and Prometheus instances. 

## Description

This PR adds support for launching [User Event service](https://github.com/apache/openwhisk/blob/master/core/monitoring/user-events/README.md) with pre configured Grafana dashboard

To give it a try download a [pre built jar](https://github.com/chetanmeh/incubator-openwhisk/releases/download/v0.13/openwhisk-standalone.jar) and then launch it as below and access http://localhost:3000 to see the dashboards

```
java -jar openwhisk-standalone.jar --user-events
```

Key aspects

1. Launches a User Event service which listens to `events` topic
2. Launches a Prometheus server with default at http://localhost:9090
3. Launches a Grafana server with default at http://localhost:3000

The Grafana server is pre-configured with various dashboards. 

```
===================================================
Launched service details

[ 3233  ] http://localhost:3233 (Controller)
[ 9092  ] localhost:9092 (kafka)
[ 9091  ] 192.168.65.2:9091 (kafka-docker)
[ 2181  ] Zookeeper (zookeeper)
[ 9000  ] http://localhost:9000 (whisk-kafka-drop-ui)
[ 3235  ] http://localhost:3235 (whisk-user-events)
[ 9090  ] http://localhost:9090 (whisk-prometheus)
[ 3000  ] http://localhost:3000 (whisk-grafana)

===================================================
```

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [x] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: core/standalone/src/main/scala/org/apache/openwhisk/standalone/UserEventLauncher.scala [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    val name = containerName("grafana")
+    val args = createRunCmd(name, env, baseParams ++ volParams)
+    val f = docker.runDetached(userEventConfig.grafanaImage, args, shouldPull = true)
+    val sc = ServiceContainer(grafanaPort, s"http://localhost:$grafanaPort", name)
+    f.map(_ => sc)
+  }
+
+  private def prometheusConfig =
+    s"""global:
+  |  scrape_interval: 10s
```
Comment (selfxp) [inline core/standalone/src/main/scala/org/apache/openwhisk/standalone/UserEventLauncher.scala]: The scrape interval definition has to be matched with the `rate` function interval (at least in the current dashboards). I wonder if we should not import this configuration from: `/compose/prometheus/prometheus.yml`

File: core/standalone/README.md [hunk 2]
```diff
@@ -245,9 +247,33 @@ be selected. TO find out the port look for message in log like below (or grep fo
...
+
+```
+java -jar openwhisk-standalone.jar --user-events
+```
+
+This mode would launch an embedded Kafka, User Event service, Prometheus and Grafana with preconfigured dashboards.
+
+```
+Launched service details
+
```
[the remark this answers is not in the document]
Comment (chetanmeh) [reply core/standalone/README.md]: Thats a good suggestion. Now added the controller to the list

```
Launched service details

[ 3233  ] http://localhost:3233 (Controller)
[ 9092  ] localhost:9092 (kafka)
[ 9091  ] 192.168.65.2:9091 (kafka-docker)
[ 2181  ] Zookeeper (zookeeper)
[ 9000  ] http://localhost:9000 (whisk-kafka-drop-ui)
[ 3235  ] http://localhost:3235 (whisk-user-events)
[ 9090  ] http://localhost:9090 (whisk-prometheus)
[ 3000  ] http://localhost:3000 (whisk-grafana)
```

File: core/standalone/src/main/scala/org/apache/openwhisk/standalone/UserEventLauncher.scala (hunk 1 above)
Comment (chetanmeh) [reply core/standalone/src/main/scala/org/apache/openwhisk/standalone/UserEventLauncher.scala]: Fixed that by packaging the existing `prometheus.yml` and then modifying it

**Comment by selfxp** [comment]
Everything looks great, thx for adding these changes @chetanmeh 👍

**Pull request merged**